### PR TITLE
1687/listing pagination issues

### PR DESCRIPTION
**Pull request by pbn4**
# Pull Request Template

## Issue

Addresses # (https://github.com/bloom-housing/bloom/issues/1687)

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

GET /listings` filters query param has been changed to support a querystring serialized array of filters, it's a breaking change because comparison property can no longer be an array. Also a property ordering problem has been resolved. Now the strict requirement for every client using the API is to use `qs` serialization format for query params.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Prototype/POC (not to merge)
- [ ] This change is a refactor/address technical debt
- [ ] This change requires a documentation update
- [ ] This change requires a SQL Script

## How Can This Be Tested/Reviewed?

Please describe the tests that you ran to verify your changes. Provide instructions so we can review. Please also list any relevant details for your test configuration

- [ ] Desktop View
- [ ] Mobile View
- [ ] Test A
- [ ] Test B

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [x] I have updated the changelog to include a description of my changes

### CODE REVIEW

File: sites/public/pages/index.tsx [hunk 1]
```diff
@@ -80,11 +81,18 @@ export default function Home(props: IndexProps) {
...
+        view: "base",
+        limit: "all",
+        $comparison: "<>",
+        status: "pending",
+      },
+      paramsSerializer: (params) => {
+        return qs.stringify(params)
+      },
+    })
+
```
Comment (seanmalbert) [inline sites/public/pages/index.tsx:95]: And again here, shouldn't there be a filter key that's an array of object? I tested with setting filter: [with different objects here] and it worked.

**Comment by dominikx96** [comment]
@seanmalbert it was because I forgot to update the backend client, can you do a review again? :)

**Review by seanmalbert — APPROVED** [review]
@pbn4 and @dominikx96 , this looks good, thanks for all the updates!

**Review by avaleske — COMMENTED** [review]
Thanks for fixing the filter ordering issue!

@seanmalbert asked me to review this PR, as well as share some other things re filters I've been thinking about.

### Questions about this refactor:
1. Can we update the name of this PR so it matches what's happening?

2. Passing an array of `ListingFilterParams` could allow for some odd edge cases. We're assuming that each ListingFilterParams will only have one comparison field and one filter field, but what if someone passes a `ListingFilterParams` that looks like this:
```
{
  $comparison: Compare["="],
   neighborhood: "Fox Creek",
   status: "pending",
},
```
   which filter field should the filter logic pick? Neighborhood or Status?

   Is there a way in the validation logic to ensure that only one field of `ListingFilterParams` is set? If so, that seems ideal, cause then we can still validate the types of each filter value. Otherwise, an array of objects with an interface like
```
{
  comparison?: Compare,
  fitler: ListingFilterKey,
  filterValue: any,
}
```
   seems less prone to error, although we'd then have to validate filter values some other way.

3. I think it'd be helpful to wrap the querystring logic in a helper function, similar to what we've done in our fork here:
https://github.com/CityOfDetroit/bloom/blob/d7ac378a7839cffd350cf993164fb68f679b8a19/ui-components/src/helpers/filters.ts#L28-L37
That way, next time the URL format changes, there's only one place to change.

### Some additional thoughts on filters that are likely out of scope for this PR
I moved these to this discussion:
https://github.com/bloom-housing/bloom/discussions/1794

**Comment by avaleske** [comment]
@seanmalbert this PR was already merged, but I shared my thoughts anyway since they point toward the future direction for filters.

File: backend/core/src/listings/dto/listing.dto.ts [hunk 2]
```diff
@@ -838,4 +861,5 @@ export const filterTypeToFieldMap: Record<keyof typeof ListingFilterKeys, string
   name: "listings.name",
   neighborhood: "property.neighborhood",
   bedrooms: "unitTypeRef.num_bedrooms",
+  leasingAgents: "leasingAgents.id",
```
Comment (avaleske) [inline backend/core/src/listings/dto/listing.dto.ts:864]: Also @seanmalbert this needed to be added to the ListingFilterKeysEnum here:
https://github.com/bloom-housing/bloom/blob/26a54444188e8c3c369a981e3b970ad419dff8f3/backend/core/src/listings/types/listing-filter-keys-enum.ts

Comment (avaleske) [reply backend/core/src/listings/dto/listing.dto.ts:864]: nvm it was! deleting this comment.

**Pull request merged**